app/containers/Scatter/saga.js

```diff
@@ -55,6 +56,31 @@ function* watchScatterConnect() {
   yield takeLatest(CONNECT_ACCOUNT, getEosAccount);
 }
 
+//
+// Refresh account data
+//
+
+function* refreshEosAccountData() {
+  const accountName = yield select(makeSelectEosAccount());
+  const eosClient = yield select(EosClient());
+  try {
+    if (accountName && accountName !== 'Attach an Account') {
+      const account = yield eosClient.getAccount(accountName);
+      const currency = yield eosClient.getCurrencyBalance('eosio.token', accountName);
+      account.currency = currency;
```

Comment:
It is usually bad practice to mutate the same object. I'd recommend the following instead: 
```
const account = {
   ...yield eosClient.getAccount(accountName),
   currency: ...yield eosClient.getCurrencyBalance('eosio.token', accountName);
}
```
The above will:
- create a new, literal object
- copy the result of the `yield eosClient.getAccount(accountName);` function (provided it's an object, please confirm).
- add a `currency` property

This way, we do everything in one go.

Reply:
Sorry I didn't explain my self: the `...` operator is called `spread` operator, it will create a shallow copy all the properties

Reply:
I'm fine with this

Reply:
I meant, i'm fine to use the "preferred method" will change tomorrow and refactor the other place that does this.